src/components/NewCommentForm/NewCommentForm.tsx

```diff
@@ -1,39 +1,100 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import { addComment } from '../../api/comments';
+import { NewCommentFormProps } from '../../types';
 import './NewCommentForm.scss';
 
-export const NewCommentForm: React.FC = () => (
-  <form className="NewCommentForm">
-    <div className="form-field">
-      <input
-        type="text"
-        name="name"
-        placeholder="Your name"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <input
-        type="text"
-        name="email"
-        placeholder="Your email"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <div className="form-field">
-      <textarea
-        name="body"
-        placeholder="Type comment here"
-        className="NewCommentForm__input"
-      />
-    </div>
-
-    <button
-      type="submit"
-      className="NewCommentForm__submit-button button"
-    >
-      Add a comment
-    </button>
-  </form>
-);
+export const NewCommentForm: React.FC<NewCommentFormProps> = ({
+  postId, fetchComments,
+}) => {
+  const [userName, setUserName] = useState('');
+  const [userEmail, setUserEmail] = useState('');
+  const [commentBody, setComment] = useState('');
+
+  const postComment = async () => {
+    await addComment({
+      name: userName,
+      email: userEmail,
+      body: commentBody,
+      postId,
+      id: 0,
+    });
+  };
+
+  const validateForm = () => {
+    // eslint-disable-next-line max-len
+    const emailRegex = /^([a-zA-Z0-9_\-.]+)@([a-zA-Z0-9_\-.]+)\.([a-zA-Z]{2,5})$/;
+    const formIsValid = userName.length && commentBody.length
+    && emailRegex.test(userEmail);
+
+    if (formIsValid) {
+      return true;
+    }
+
+    return false;
+  };
```

Comment:
add some error displaying, when email is incorrect. Not user friendly now

Reply:
Done, thx
I kinda hardcoded this fix, will refactor it later